activerecord/lib/active_record/pattern_matching.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  # = Active Record Pattern Matching
+  module PatternMatching
+    # Provides the pattern matching interface for an Active Record model.
+    module Record
+      # Returns a hash of attributes for the given keys. Provides the pattern
+      # matching interface for matching against hash patterns. For example:
+      #
+      #   class Person < ActiveRecord::Base
+      #   end
+      #
+      #   def greeting_for(person)
+      #     case person
+      #     in { name: "Mary" }
+      #       "Welcome back, Mary!"
+      #     in { name: }
+      #       "Welcome, stranger!"
+      #     end
+      #   end
+      #
+      #   person = Person.new
+      #   person.name = "Mary"
+      #   greeting_for(person) # => "Welcome back, Mary!"
+      #
+      #   person = Person.new
+      #   person.name = "Bob"
+      #   greeting_for(person) # => "Welcome, stranger!"
+      def deconstruct_keys(keys)
```

Comment:
(Sorry, just passing by.)

Note that `keys` might be `nil`, if somebody does this:
```ruby
case person
in name: 'Mary', **attributes
  Rails.logger.info "Mary logged in. Currently set of attrs are: #{attributes}"
end
```

Reply:
Yeah, I handled that in the gem I mentioned. I wasn't sure if this PR was moving anywhere so I'm waiting to hear on this.

Reply:
I'm not doing anything fancy, just destructuring rightward assignment, and the gem is working great for that purpose!
```
    rec = MyRecord.find(id)

    # Destructuring Rightward Assignment
    # See: https://www.fullstackruby.dev/ruby-3-fundamentals/2021/01/06/everything-you-need-to-know-about-destructuring-in-ruby-3/
    rec => {
      organization_id:,
      group_id:,
      source_id:,
      patch_id:,
      report_id:,
    }
    # and now I have all those local variables to work with, pulled directly from my recorrds' attributes.
```

I have noticed that it only works with true attributes, so if I create an accessor method I can't pull that value through (yet).

Reply:
@pboling that's expected. If you have any other feedback on the gem though feel free to open an issue on the gem repo itself